nmpolicy/internal/resolver/replace.go

```diff
@@ -21,7 +21,7 @@ import (
 )
 
 func replace(inputState map[string]interface{}, path ast.VariadicOperator, expectedNode ast.Node) (map[string]interface{}, error) {
-	replaced, err := applyFuncOnPath(inputState, path, expectedNode, replaceMapFieldValue, true)
+	replaced, err := applyFuncOnMap(path, inputState, expectedNode, replaceMapFieldValue, true, true)
```

Comment:
I think that in case of a replace we shouldn't filter anything. The output should similar to the input + the replacement.
When you do find and replace on a doc, you don't assume the docs will be filtered as well.